lib/puppet/defaults.rb

```diff
@@ -249,7 +249,7 @@ def self.default_diffargs
         :desc     => "Whether to create the necessary user and group that puppet agent will run as.",
     },
     :manage_internal_file_permissions => {
-        :default  => true,
+        :default  => ! Puppet.features.microsoft_windows?,
```

Comment:
I thought we had to load settings before features could be evaluated, so I'd expect this to be in a `lambda { ! Puppet.features.microsoft_windows? }`. Or even better make it conditional on `Puppet::Util::Platform.windows?`, so you don't have to mess with the feature logic.

Reply:
Copy-pasta from other examples of it e.g.

https://github.com/puppetlabs/puppet/blob/4.10.x/lib/puppet/defaults.rb#L1274-L1278

Reply:
Yep, that's it.

Reply:
Ha...serious fail there.  That was the ONLY example where it used a lamba. I'll change that.

Reply:
Fixed

Reply:
Was there a reason we couldn't use [`Puppet::Util::Platform.windows?`](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/platform.rb#L7)

Reply:
I used features instead `Puppet::Util::Platform.windows?` due to some historical issues with mocking that call.  That said, I'm not really fussed either way.